src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScore.java

```diff
@@ -40,19 +48,19 @@ public ScoreValue calculate(Value<?>... values) {
           .withMinConfidence();
     }
 
-    if (artifactVersions.get().size() <= 1) {
+    if (artifactVersions.get().size() <= ARTIFACT_VERSIONS_SIZE_THRESHOLD) {
       return scoreValue(0.0, artifactVersions)
           .explain("Only one version is given. Hence, no release history score can be calculated")
           .withMinConfidence();
     }
 
-    ScoreValue scoreValue = scoreValue(7.0, artifactVersions);
+    ScoreValue scoreValue = scoreValue(6.0, artifactVersions, artifactVersion);
```

Comment:
I am not sure I understand why the score should be lower if a project doesn't use semantic versions. Is it semantic versioning any better than other schemes?

I am not sure either why the score should be lower when release history for a major version is short. It can happen that a project has a long release history but it just recently shipped a new major version. In this case, I'd fallback to using the entire release history.

@mibo What do you think?

Reply:
As per my understanding. If we are calculating the score on more related Artifacts as compared to all the artifacts. This would avoid variations and would handle diversity better. I believe the more focused related list of artifacts should have a better score as compared to considering all the artifacts (which may or may not be related). The idea is to have an appropriate clustering before we calculate the score.

Reply:
> I am not sure either why the score should be lower when release history for a major version is short. It can happen that a project has a long release history but it just recently shipped a new major version. In this case, I'd fallback to using the entire release history.

Afaik the current implementation is that in case of only one version for a major version exists (e.g. only `2.0.0`) it fallback to use the entires release history.

However, maybe the score can calculated the release trend for all versions and for the according major version and build a final trend out of both (just as an idea).

> I am not sure I understand why the score should be lower if a project doesn't use semantic versions. Is it semantic versioning any better than other schemes?

Idea was that if we have the semantic version that only the versions with the same major version are considered (the clustering mentioned by Sourabh).
As sample, someone want the score for version `1.2.0` (release one year after `1.1.0`) but the latest is `3.1.0` (release one month before 3.0.0) the score should be lower.
If the version is not a semantic versions this may lead to higher scores why we thought to start with a lower score value (`6`).

Reply:
I agree with @mibo. This way the focus would be more on the possible affected list of versions.

Reply:
> Afaik the current implementation is that in case of only one version for a major version exists (e.g. only 2.0.0) it fallback to use the entires release history.

@sourabhsparkala Do we have a test case for that?

> Idea was that if we have the semantic version that only the versions with the same major version are considered (the clustering mentioned by Sourabh).

@sourabhsparkala Do we have a test case for that? Do we have test cases with not-semantic versions?

Reply:
We do not have any test cases for non-sematic. As `SemanticVersion` would fail. We have not figured a way to handle non-sematic as of now. This issue is #480

Reply:
> We do not have any test cases for non-sematic. As `SemanticVersion` would fail. We have not figured a way to handle non-sematic as of now. This issue is #480

The current scoring should not fail if it sees a non-semantic version. #480 is about improving the scoring function to handle not-semantic versions in a better way. I think it would be good to add tests for the statements from @mibo above. We'll see then what kind of scores the function produces.